Integration tests are failing

Over in #1068 I added two integration tests, so I actually ran them and we're getting a whole host of failures. None of them are related to what I'm doing there -- the new tests are passing fine -- so I'm going to ignore it in that PR, but we should patch these up.

```
======================================================================
ERROR: test_Fe_ferro_C (tests_sphinx_sphinx_check_all.TestSphinx)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/test_integration/tests_sphinx_sphinx_check_all.py", line 73, in test_Fe_ferro_C
    job.structure.set_initial_magnetic_moments([2, 2])
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/atoms.py", line 2631, in set_initial_magnetic_moments
    raise ValueError("magmoms can be collinear or non-collinear.")
ValueError: magmoms can be collinear or non-collinear.

======================================================================
ERROR: test_Fe_ferro_constraint (tests_sphinx_sphinx_check_all.TestSphinx)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/test_integration/tests_sphinx_sphinx_check_all.py", line 125, in test_Fe_ferro_constraint
    job.structure.set_initial_magnetic_moments([2, 2])
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/atoms.py", line 2631, in set_initial_magnetic_moments
    raise ValueError("magmoms can be collinear or non-collinear.")
ValueError: magmoms can be collinear or non-collinear.

======================================================================
ERROR: test_Fe_nonmag (tests_sphinx_sphinx_check_all.TestSphinx)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/test_integration/tests_sphinx_sphinx_check_all.py", line 61, in test_Fe_nonmag
    job.structure.set_initial_magnetic_moments([2, 2])
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/atoms.py", line 2631, in set_initial_magnetic_moments
    raise ValueError("magmoms can be collinear or non-collinear.")
ValueError: magmoms can be collinear or non-collinear.

======================================================================
ERROR: test_sxextopt_Fe (tests_sphinx_sphinx_check_all.TestSphinx)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/test_integration/tests_sphinx_sphinx_check_all.py", line 166, in test_sxextopt_Fe
    spx.structure.set_initial_magnetic_moments([2, 2])
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/pyiron_atomistics/atomistics/structure/atoms.py", line 2631, in set_initial_magnetic_moments
    raise ValueError("magmoms can be collinear or non-collinear.")
ValueError: magmoms can be collinear or non-collinear.

======================================================================
FAIL: test_Al_minimize (tests_sphinx_sphinx_check_all.TestSphinx)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyiron_atomistics/pyiron_atomistics/test_integration/tests_sphinx_sphinx_check_all.py", line 106, in test_Al_minimize
    self.assertGreater(E[0], E[1], 'Energy not decreased')
AssertionError: -57.19326599140298 not greater than -57.19326592068059 : Energy not decreased

----------------------------------------------------------------------
```

--- a/test_integration/tests_sphinx_sphinx_check_all.py
+++ b/test_integration/tests_sphinx_sphinx_check_all.py
@@ -98,7 +98,7 @@ class TestSphinx(unittest.TestCase):
 
     def test_Al_minimize(self):
         job = self.project.create_job(self.project.job_type.Sphinx, 'spx_Al_minimize')
-        job.structure = self.project.create.structure.ase.bulk('Al', a=self.a_Al)
+        job.structure = self.project.create.structure.ase.bulk('Al', a=self.a_Al, cubic=True)
         job.structure.positions[0, 0] += 0.01
         job.calc_minimize()
         job.run()

--- a/test_integration/tests_sphinx_sphinx_check_all.py
+++ b/test_integration/tests_sphinx_sphinx_check_all.py
@@ -103,7 +103,7 @@ class TestSphinx(unittest.TestCase):
         job.calc_minimize()
         job.run()
         E = job['output/generic/energy_tot']
-        self.assertGreater(E[0], E[1], 'Energy not decreased')
+        self.assertGreater(E[0], E[-1], 'Energy not decreased')
 
     def test_check_overlap(self):
         job = self.project.create_job(self.project.job_type.Sphinx, 'spx_check_overlap')

--- a/test_integration/tests_sphinx_sphinx_check_all.py
+++ b/test_integration/tests_sphinx_sphinx_check_all.py
@@ -58,7 +58,7 @@ class TestSphinx(unittest.TestCase):
         )
         job = self.project.create_job(self.project.job_type.Sphinx, 'spx_Fe_ferro')
         job.structure = self.project.create.structure.ase.bulk('Fe', a=self.a_Fe)
-        job.structure.set_initial_magnetic_moments([2, 2])
+        job.structure.set_initial_magnetic_moments(len(job.structure) * [2])
         job.calc_static()
         job.run()
         self.assertLess(
@@ -70,7 +70,7 @@ class TestSphinx(unittest.TestCase):
     def test_Fe_ferro_C(self):
         job = self.project.create_job(self.project.job_type.Sphinx, 'spx_Fe_ferro_C')
         job.structure = self.project.create.structure.ase.bulk('Fe', a=self.a_Fe)
-        job.structure.set_initial_magnetic_moments([2, 2])
+        job.structure.set_initial_magnetic_moments(len(job.structure) * [2])
         job.structure += self.project.create_atoms(
             elements=['C'], positions=[[0, 0, 0.5 * self.a_Fe]], magmoms=[0]
         )
@@ -122,7 +122,7 @@ class TestSphinx(unittest.TestCase):
     def test_Fe_ferro_constraint(self):
         job = self.project.create_job(self.project.job_type.Sphinx, 'spx_Fe_ferro_constraint')
         job.structure = self.project.create.structure.ase.bulk('Fe', a=self.a_Fe)
-        job.structure.set_initial_magnetic_moments([2, 2])
+        job.structure.set_initial_magnetic_moments(len(job.structure) * [2])
         job.fix_spin_constraint = True
         job.calc_static()
         job.run()
@@ -163,7 +163,7 @@ class TestSphinx(unittest.TestCase):
     def test_sxextopt_Fe(self):
         spx = self.project.create_job('Sphinx', 'spx_sxextopt_Fe')
         spx.structure = self.project.create.structure.ase.bulk('Fe', a=2)
-        spx.structure.set_initial_magnetic_moments([2, 2])
+        spx.structure.set_initial_magnetic_moments(len(spx.structure) * [2])
         spx.server.run_mode.interactive = True
         spx.calc_static()
         sxextopt = self.project.create_job('SxExtOptInteractive', 'sxextopt_Fe')

